README.md

```diff
@@ -82,16 +161,130 @@ default.
 
 E.g. to build with the current system's config file run:
 
-    $ skt.py --workdir skt-workdir -vv \
+    $ skt.py --rc skt-rc --state --workdir skt-workdir -vv \
              build -c /boot/config-`uname -r`
 
-All the above commands use the `-vv` option to increase verbosity of the
-command's output, so it's easier to debug problems. Remove the option for
-quiter, shorter output.
+### Publish
+
+To "publish" the resulting build using the simple "cp" (copy) publisher run:
+
+    $ skt.py --rc <SKTRC> --state --workdir <WORKDIR> -vv \
+             publish -p cp <DIRECTORY> <URL_PREFIX>
+
+Here `<DIRECTORY>` would be the location for the copied build artifacts, and
+`URL_PREFIX` would be the string to add to prepend the filenames with
+(together with a slash `/`) to construct the URLs the files will be reachable
+at. The resulting URLs will be passed to other commands, such as `run`, via
+the saved state in the configuration file.
+
+E.g. to publish to the `/srv/builds` directory available at
+`http://skt-server` run:
+
+    $ skt.py --rc skt-rc --state --workdir skt-workdir -vv \
+             publish -p cp /srv/builds http://skt-server
+
+### Run
+
+To run the tests you will need access to a
+[Beaker](https://beaker-project.org/) instance configured to the point where
+`bkr whoami` completes successfully. You will also need a template file for
+generating a Beaker job XML, which runs the tests. The template file can
+contain the following placeholder strings replaced by `skt` before submitting
+the job XML to Beaker:
+
+* `##KVER##`
+    - The kernel release version output by `make -s kernelrelease`.
+* `##KPKG_URL##`
+    - The URL of the kernel tarball, generated and published to with
+      `publish`.
+* `##UID##`
+    - A string of test job "tags" generated by `merge` reset by `run` on
+      baseline checks. For use in `<whiteboard>`.
+* `##HOSTNAME##`
+    - A decorated name of the host the job runs on, or empty string if no
+      specific host was picked. For use in `<whiteboard>`.
+* `##HOSTNAMETAG##`
+    - An XML fragment limiting the job to a specific host, or empty string if
+      no specific host was picked. For use in `<hostRequires>`.
```

Comment:
It'd be great if you follow the same approach than above and put an example of the template here. Honestly I didn't get it at first, tomorrow I'll read it again and see if I get it.

Reply:
Good idea. I added the example. However, I had to strip quite a lot of stuff from it, so it won't work (not that it would work outside Red Hat anyway), but it can serve as an illustration. Hope that helps.